src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -318,7 +318,7 @@ TOKEN : /* Numeric Constants */
             )>
   |     < S_LONG: ( <DIGIT> )+ >
   |     < #DIGIT: ["0" - "9"] >
-  |     < S_HEX: ("x'" ( <HEX_VALUE> )+ "'" | "0x" ( <HEX_VALUE> )+ ) >
+  |     < S_HEX: (["x","X"] "'" ( <HEX_VALUE> )+ "'" | "0x" ( <HEX_VALUE> )+ ) >
```

Comment:
Since javacc builds case insensitive parser, this change is not necessary. I also included a test for this and it succeeded.